tests/CMakeLists.txt

```diff
@@ -0,0 +1,271 @@
+## ---------------------------------------------------------------------
+##
+## Copyright (c) 2020 - 2020 by the IBAMR developers
+## All rights reserved.
+##
+## This file is part of IBAMR.
+##
+## IBAMR is free software and is distributed under the 3-clause BSD
+## license. The full text of the license can be found in the file
+## COPYRIGHT at the top level directory of IBAMR.
+##
+## ---------------------------------------------------------------------
+
+ADD_CUSTOM_TARGET(tests)
```

Comment:
The test setup seems more compact than the examples setup.

It seems like most of the lines in this patch are setting up all of the examples. 😀

Reply:
You're absolutely right. It's almost as if I had this in mind when I wrote the test suite :)

Reply:
Is it worth taking the time now to rationalize the layout of the examples?

Reply:
Or do you think it is better to allow examples to be more heterogeneous?

Reply:
I'm not sure this is even possible. We have, for example:
- mesh generation programs that have no output and don't link with IBAMR
- examples with external dependencies
- examples with multiple input files
- extra executables that do convergence testing
- some examples require libMesh, others do not
- examples with multiple source files

none of these are amendable to non-custom treatment. I think what we have is fine.